Ginger/GingerCore/ALM/QtestCore.cs

```diff
@@ -0,0 +1,657 @@
+
+            string runfieldInRestSyntax = QtestConnect.Instance.ConvertResourceType(ALM_Common.DataContracts.ResourceType.TEST_RUN);
+            List<QTestApiModel.FieldResource> runfieldsCollection = fieldApi.GetFields((long)Convert.ToInt32(ALMCore.DefaultAlmConfig.ALMProjectKey), runfieldInRestSyntax);
+
+            fields.Append(AddFieldsValues(testSetfieldsCollection, testSetfieldInRestSyntax));
+            fields.Append(AddFieldsValues(testCasefieldsCollection, testCasefieldInRestSyntax));
+            fields.Append(AddFieldsValues(designStepfieldsCollection, designStepfieldInRestSyntax));
+            fields.Append(AddFieldsValues(testInstancefieldsCollection, testInstancefieldInRestSyntax));
+            //fields.Append(AddFieldsValues(designStepParamsfieldsCollection, designStepParamsfieldInRestSyntax));
+            fields.Append(AddFieldsValues(runfieldsCollection, runfieldInRestSyntax));
+
+            return fields;
+        }
+        private  ObservableList<ExternalItemFieldBase> AddFieldsValues(List<QTestApiModel.FieldResource> testSetfieldsCollection, string testSetfieldInRestSyntax)
+        {
+            //TODO: need to handle duplicate fields
+            ObservableList<ExternalItemFieldBase> fields = new ObservableList<ExternalItemFieldBase>();
+
+            if ((testSetfieldsCollection != null) && (testSetfieldsCollection.Count > 0))
+            {
+                foreach (QTestApiModel.FieldResource field in testSetfieldsCollection)
+                {
+                    if (string.IsNullOrEmpty(field.Label)) continue;
+
+                    ExternalItemFieldBase itemfield = new ExternalItemFieldBase();
+                    //itemfield.ID = field.OriginalName;
+                    itemfield.ID = field.Id.ToString();
+                    itemfield.ExternalID = field.OriginalName;  // Temp ??? Check if ExternalID has other use in this case
+                    itemfield.Name = field.Label;
+                    bool isMandatory;
+                    bool.TryParse(field.Required.ToString(), out isMandatory);
+                    itemfield.Mandatory = isMandatory;
+                    bool isSystemField;
+                    bool.TryParse(field.SystemField.ToString(), out isSystemField);
+                    itemfield.SystemFieled = isSystemField;
+                    if (itemfield.Mandatory)
+                        itemfield.ToUpdate = true;
+                    itemfield.ItemType = testSetfieldInRestSyntax.ToString();
+                    itemfield.Type = field.DataType;
+
+                    if (field.AllowedValues != null)
+                    {
+                        foreach (QTestApiModel.AllowedValueResource value in field.AllowedValues)
+                        {
+                            itemfield.PossibleValues.Add(value.Label);
+                            //itemfield.PossibleValues.Add(value.Value.ToString());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)